plugins/providers/hyperv/scripts/configure_vm.ps1

```diff
@@ -125,3 +127,17 @@ try {
     Write-ErrorMessage "Failed to ${AutoAction} automatic checkpoints on VM: ${PSItem}"
     exit 1
 }
+
+if($VM.Generation -gt 1) {
+    try {
+        if($EnableTrustedPlatformModule) {
+            Hyper-V\Enable-VMTPM -VM $VM
+            $tpmAction = "enable"
```

Comment:
this should be done before calling the `Hyper-V\Enable-VMTPM` cmdlet. Or else the `catch` block will try to use an uninitialized variable.

Reply:
thanks for catching that!